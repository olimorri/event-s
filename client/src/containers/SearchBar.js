/* eslint-disable no-unused-vars */
import React, { useState } from 'react';
import { Flex, Box, FormControl, Input } from '@chakra-ui/react';

export default function SearchBar({ searchTerm, setSearchTerm }) {
  const handleChange = (event) => {
    setSearchTerm(event.target.value.toLowerCase());
  };

  return (
    <Box>
      <FormControl>
        <Flex>
          <FormControl
            minWidth={'30vh'}
            w={'60vh'}
            id="search_bar"
            marginRight={4}
          >
            <Input
              id="search"
              placeholder="Search for events"
              value={searchTerm}
              name="search"
              onChange={handleChange}
              bg={'white'}
              w={'100%'}
            />
          </FormControl>
        </Flex>
      </FormControl>
    </Box>
  );
}
