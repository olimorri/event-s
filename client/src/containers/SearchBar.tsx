/* eslint-disable no-unused-vars */
import React, { ChangeEvent } from 'react';
import { Flex, Box, FormControl, Input } from '@chakra-ui/react';

export default function SearchBar({
  searchTerm,
  setSearchTerm,
}: {
  searchTerm: string;
  setSearchTerm: Function;
}) {
  const handleChange = (e: ChangeEvent<HTMLInputElement>): void => {
    setSearchTerm(e.target.value.toLowerCase());
  };

  return (
    <Box>
      <FormControl>
        <Flex>
          <FormControl
            minWidth={'30vh'}
            w={'60vh'}
            id="search_bar"
            marginRight={4}
          >
            <Input
              id="search"
              placeholder="Search for events"
              value={searchTerm}
              name="search"
              onChange={handleChange}
              bg={'white'}
              w={'100%'}
            />
          </FormControl>
        </Flex>
      </FormControl>
    </Box>
  );
}
