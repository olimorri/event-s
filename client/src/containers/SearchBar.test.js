import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import SearchBar from './SearchBar';

const SearchTest = () => (
  <BrowserRouter>
    <SearchBar />
  </BrowserRouter>
);

describe('Snapshot', () => {
  it('should take a snapshot', () => {
    const { asFragment } = render(<SearchTest />);
    expect(asFragment(<SearchTest />)).toMatchSnapshot();
  });
});

describe('Unit Testing', () => {
  it('should render the input', () => {
    render(<SearchTest />);
    const SearchInput = screen.getByPlaceholderText(/Search for events/i);
    expect(SearchInput).toBeInTheDocument();
  });
});
