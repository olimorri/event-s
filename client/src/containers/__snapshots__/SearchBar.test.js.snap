// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-0"
  >
    <div
      class="chakra-form-control css-1kxonj9"
      role="group"
    >
      <div
        class="css-k008qs"
      >
        <div
          class="chakra-form-control css-s5jkx7"
          role="group"
        >
          <input
            class="chakra-input css-1so4oux"
            id="search"
            name="search"
            placeholder="Search for events"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
